packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -146,23 +156,50 @@ export function entityDeserializer(edmToTs, extractODataETag) {
     }
   }
 
-  function deserializeComplexType<EntityT extends EntityBase>(
+  function deserializeComplexTypeLegacy<
```

Comment:
please add a todo and mark it as deprecated so we don't ignore it when doing sdk v2.

Reply:
check